src/Microsoft.Win32.Primitives/ref/Microsoft.Win32.Primitives.cs

```diff
@@ -8,13 +8,15 @@
 
 namespace System.ComponentModel
 {
-    public partial class Win32Exception : System.Exception
+    public partial class Win32Exception : System.Runtime.InteropServices.ExternalException
```

Comment:
Exception itself does, and then provides a virtual GetObjectData member for derived exceptions to override.  For whatever reason, some Exception-derived types also choose to implement ISerializable again, but it's not necessary, and this one doesn't in desktop.